### PR TITLE
Add gateway support

### PR DESCRIPTION
**Pull request by kraman**
* Add gateway model and CRUD hooks
* Add gateway create/list/destroy Mesh Admin commands
* Add events to track process exit

Connect to strongloop-internal/scrum-nodeops#674

### CODE REVIEW

**Comment by sam-github** [comment]
@slnode test please

**Comment by kraman** [comment]
@slnode test please

**Comment by kraman** [comment]
Im trying to figure out why express metrics test is failing

File: bin/sl-meshadm.txt [hunk 1]
```diff
@@ -22,3 +22,12 @@ Executor commands:
   exec-remove EXEC        Remove executor EXEC.
   exec-list               List executors.
   exec-shutdown EXEC      Shutdown executor EXEC.
+
+Gateway controller commands:
+
+`GW` is the gateway controller ID. It can be obtained by listing gateway
+controllers using `gw-list`.
+
+  gw-create             Create a gateway controller.
```
Comment (sam-github) [inline bin/sl-meshadm.txt]: OCD alert: I'd indent the descriptive text the same as the previous block.

File: client/model-config.json [hunk 2]
```diff
@@ -68,5 +68,9 @@
   "InstanceAction": {
     "dataSource": "remote",
     "public": false
+  },
+  "Gateway": {
+    "dataSource": "remote",
+    "public": false
```
Comment (sam-github) [inline client/model-config.json]: executors are public, shouldn't this be, too? so they can be directly created by clients (the installers).

File: server/models/service-process.js [hunk 3]
```diff
@@ -43,7 +44,10 @@ module.exports = function extendServiceProcess(ServiceProcess) {
         return proc.updateAttributes({
           stopTime: new Date(),
           stopReason: pInfo.reason,
-        }, asyncCb);
+        }, function(err, proc) {
+          if (err) return asyncCb(err);
+          serviceManager.onProcessExit(proc, asyncCb);
```
Comment (sam-github) [inline server/models/service-process.js]: I don't see the relationship between any of the service-process changes and gateway, are they an unrelated bug fix?

**Comment by sam-github** [comment]
A couple small comments, but otherwise LGTM

File: client/model-config.json (hunk 2 above)
Comment (kraman) [inline client/model-config.json]: this being the client side, the public field doesnt make any difference but Ive updated anyway

File: server/models/service-process.js (hunk 3 above)
Comment (kraman) [inline server/models/service-process.js]: The link between the two is set up in strong-central

**Comment by kraman** [comment]
@slnode test please

**Comment by kraman** [comment]
@slnode test please

**Comment by kraman** [comment]
@slnode test please
